_source/_posts/2022-01-13-spring-cloud-contract.md

```diff
@@ -0,0 +1,893 @@
+
+You'll use the Spring Initializr project to bootstrap the Producer project. Open [this link](https://start.spring.io/#!type=maven-project&language=java&platformVersion=2.6.2&packaging=jar&jvmVersion=11&groupId=com.example&artifactId=producer&name=producer&description=Producer%20project%20for%20Spring%20Boot%20Contract&packageName=com.example.contract&dependencies=web,cloud-contract-verifier,lombok) to access the pre-configured starter project.
+
+{% img blog/spring-cloud-contract/producer-starter-config.png alt:"Producer Starter Configuration" width:"1000" %}{: .center-image }
+
+You are creating a Java project -- Java 11 -- using Maven as the build system. The Spring Boot version was 2.6.2 when I wrote this tutorial. You are including three dependencies: 
+
+- **[Spring Web](https://docs.spring.io/spring-framework/docs/current/reference/html/web.html)**: adds the basic HTTP/web capabilities to build a REST API 
+- **[Contract Verifier](https://spring.io/projects/spring-cloud-contract)**: brings in Spring Cloud Contract project by enabling Consumer Drive Contracts and stub generation
+- **[Lombok](https://projectlombok.org/)**: helper project that will generate some getters, setters, and constructors
+
+When you're ready, click **Generate** and save the zip file in an appropriate parent directory. You'll be making two Spring Boot projects (as mentioned, a producer and a consumer), so it probably makes sense to have a parent directory named something like `Spring Cloud Contract Tutorial` or `spring-cloud-contract-tutorial` (if you dislike spaces in directory names). 
+
+Unzip the file and open the directory in your favorite editor or IDE (I'm an IDE/IntelliJ man myself -- hopefully that's not disappointing to y'all -- because in the immortal words of Buckaroo Banzai "Life is too short for vim.").
+
+Let's start by building the hat producer. Both the producer and consumer projects will be Spring Boot resource servers that use Spring Web to implement a simple REST API.
+
+Create a `Hat.java` file that is the data model for the hats the producer will serve.
+
+`src/main/java/com/example/contract/Hat.java`
+
+```java
+package com.example.contract;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class Hat {
+    private Long id;
+    private String name;
+    private Long size;
+    private String color;
+}
+```
+
+The hat has four attributes, an `id` as well as `name`, `size`, and `color`. Notice the two annotations. These are what Lombok helped us with. The `@Data` annotation generates getters and setters as well as a `toString()` and a constructor for all required arguments ([check out the docs](https://projectlombok.org/features/Data)). The `@AllArgsConstructor` annotation generates an constructor for all fields in the class. I really like it because it de-clutters the data model classes and saves some time generating all of the 'ceremony' methods. 
```

Comment:
```suggestion
The hat has four attributes, an `id` as well as `name`, `size`, and `color`. Notice the two annotations. These are what Lombok helped us with. The `@Data` annotation generates getters and setters as well as a `toString()` and a constructor for all required arguments ([check out the docs](https://projectlombok.org/features/Data)). The `@AllArgsConstructor` annotation generates a constructor for all fields in the class. I really like it because it de-clutters the data model classes and saves some time generating all of the 'ceremony' methods. 
```
typo